### PR TITLE
Adapt high-availability configuration

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area high-availability
/kind enhancement

**What this PR does / why we need it**:
This PR prepares components for HA by labeling them with the corresponding `high-availability-config.resources.gardener.cloud/type` and defining the required `PodDisruptionBudget`, please see [this document](https://github.com/gardener/gardener/blob/master/docs/development/high-availability.md) for more information.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6529

**Release note**:

```feature operator
Extension and control-plane components have been adjusted to the high availability configuration supported by Gardener as of release `v1.60`. See [this document](https://github.com/gardener/gardener/blob/master/docs/development/high-availability.md) for more information about HA in gardener.
```
```feature user
This release adds HA support for the service's control-plane components when extension is deployed along with Gardener >= `v1.60`. Please see [this document](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_high_availability.md) for more information about HA settings in shoot clusters.
```

### CODE REVIEW

File: example/controller-registration.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ metadata:
...
...Axszvsn64Md21P6ktKn/L3PkkfdmM4G19z8fFe//O8bXO///HlJR56WarTCkPq3F93vTffz6G2RQnl3X3lsGzLyldsj+2ppj0/G/8OnG3wFfu/5biv/tHh93d+P/PlJhyx+7V34hq7Bj10ZRDw1q453fqRuE23fdtlQZN3iYA0pceWZfFmbBtppDA1Iaao9k66yiLYkXqYbg5fOEyknjBnmmpYZfpK9ENMXek+xIhWO5fXEcRY1pMPy45qNdZYr0DIUOcNWf6loPpgSwYHSt4wXLrUgQoNtQISHLH2qC0TVTTrdWxZpjfinePIsOaaq9hUOv9qrIEQEiSkwZtkQtrGfl/rL39P9ZqZn+l9cvbqr4ZVqj/w8P90+K33/t7r7/fj9JBDpzrZh8G6xHWKzPjAA1fCoefuDhnCv7HLa6fFg1GiM662X3fvpKjPTo9tKLrsQdlq1WPk60p+hnDBXtJYsirWw9AM9utWAMY9n0A5Aypu5BhXZ/UNLUDzI9fbC/76Q5ieY86H51Ya3VKG00cu1WS4nExUKFWOYeOYbM2gjdxElEnzWNh05tUhlRuprTExHIVeHJPXK4T4Cu4jfxRNNXfdbzgAfi575i2JPVJB8WFawWjOZZ6X07yaFEL7nvTpPX6nm34ouv4s5e3KcULyp0vLARymlp+UHOX725C1Dsn/IT6XgnjiyRfCYzuYgUobXFflcHU6PbLAIxM4KlH+leMOuYVsAjacQ9Nw8qP0R5sh/Cm+vrc76hfXY52T9Qr9RIIZWvQsLvF/JCSX7TFwKes2gvJENReXJHXAINjgWmb4fPR5dE/B1P+uRqPLrpXw/JD8NX/H1aWtcLkMPLs2oI7OPKT0HKC4ee4t8j/YgXLH6mERuS3IwpV7vxbLciSSG5mzOXxHj5eHorT3JR6cIyeRc+yDVvMBxfj74bDYDMpFEPkhY9UJpTVe7DIWsRUvx6rKI++CAEBOA+ong/FceM+e2F4mD7MlndJJYrzo4jsmwvp7Xum5IqclFtKC5R1TQusRm0RvMfrIQKaz6d0WqVTxb0yE8/t1oPSFXwLn4kBSVIhI73+HMuvjj77kZLaEShS8eKbZhZ0Tye4vjLbEE1bEOLUtc2gYSbkfa+3tW/BlWbAPda6b32PdyB+NhWdJd2aZd2aZd2aZd2aZd2aZd2aZd26a+d/g9HxTdpAKAAAA==
```
[the remark this answers is not in the document]
Comment (MartinWeindel) [reply example/controller-registration.yaml:8]: Running helm on Linux results in different byte stream...

**Review by MartinWeindel — APPROVED** [review]
/lgtm

File: charts/internal/shoot-cert-management-seed/templates/poddisruptionbudget.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{{- if gt (int .Values.replicaCount) 1 }}
```
Comment (MartinWeindel) [inline charts/internal/shoot-cert-management-seed/templates/poddisruptionbudget.yaml:1]:
```suggestion
```

@timuthy Is this line too much?

**Review by MartinWeindel — COMMENTED** [review]
Can you check if the suggested change is your intention?

File: charts/internal/shoot-cert-management-seed/templates/poddisruptionbudget.yaml (hunk 2 above)
Comment (timuthy) [reply charts/internal/shoot-cert-management-seed/templates/poddisruptionbudget.yaml:1]: Yes, thanks for noticing. Copy-paste issue 🙃

**Review by MartinWeindel — COMMENTED** [review]
next try to fix the controller-registration.yaml

**Review by MartinWeindel — APPROVED** [review]
/lgtm

**Comment by timuthy** [comment]
Thanks @MartinWeindel

**Pull request merged**